### ISSUE TITLE
ml: LSTM trains on entire dataset as one full batch — no mini-batching, gradient instability

### ISSUE THREAD

**Issue by f1xgun** [f1xgun/finalayze#140]
**File:** src/finalayze/ml/models/lstm_model.py:130-136

**Issue:** The training loop passes the full `x_tensor` (potentially thousands of sequences) as a single batch to the optimizer:

```python
for _ in range(_TRAIN_EPOCHS):
    optimizer.zero_grad()
    output = self._model(x_tensor)
    loss = criterion(output, y_tensor)
    loss.backward()
    optimizer.step()
```

For a 2-year daily dataset of 3 symbols, this yields ~1400 sequences in a single batch. Problems:

1. **Memory:** Full-batch BPTT on thousands of sequences can OOM on CPU with longer hidden sizes.
2. **Gradient quality:** Full-batch gradient descent has poor generalisation compared to mini-batch SGD; the loss landscape is smooth but narrow minima are common.
3. **No learning-rate schedule:** Fixed LR of 0.001 with Adam for 50 epochs on a full batch is unlikely to converge to a good minimum.

**Fix:** Implement mini-batch training with a DataLoader (batch size 32–64), add a learning-rate scheduler (ReduceLROnPlateau), and add early stopping on a validation split.
